pkg/rpcclient/wsclient.go

```diff
@@ -711,7 +711,12 @@ func (c *WSClient) makeWsRequest(r *neorpc.Request) (*neorpc.Response, error) {
 
 func (c *WSClient) performSubscription(params []any, rcvr notificationReceiver) (string, error) {
 	var resp string
-
+	if rcvr.Filter() != nil {
+		err := rcvr.Filter().IsValid()
+		if err != nil {
+			return "", err
+		}
+	}
```

Comment:
Good, I see it's here. You may simplify this check by a single `if flt := rcvr.Filter(); flt != nil {` and then use `flt` inside `if`.